libraries/launcher/org/prismlauncher/EntryPoint.java

```diff
@@ -52,113 +54,126 @@
-                    throw new ParseException("Error while parsing:" + inData);
+            return ExitCode.ABORT;
+        }
 
-                params.add(tokens[0], tokens[1]);
+        if (action == PreLaunchAction.ABORT) {
+            Log.fatal("Launch aborted by the launcher");
 
-                return Action.Proceed;
-            }
+            return ExitCode.ABORT;
         }
-    }
 
-    public int listen() {
-        Action action = Action.Proceed;
+        try {
+            Launcher launcher;
+            String type = params.getString("launcher");
 
-        try (BufferedReader reader = new BufferedReader(new InputStreamReader(
-                System.in,
-                StandardCharsets.UTF_8
-        ))) {
-            String line;
+            switch (type) {
+                case "standard":
+                    launcher = new StandardLauncher(params);
+                    break;
 
-            while (action == Action.Proceed) {
-                if ((line = reader.readLine()) != null) {
-                    action = parseLine(line);
-                } else {
-                    action = Action.Abort;
-                }
+                case "legacy":
+                    launcher = new LegacyLauncher(params);
+                    break;
+
+                default:
+                    throw new IllegalArgumentException("Invalid launcher type: " + type);
             }
-        } catch (IOException | ParseException e) {
-            LOGGER.log(Level.SEVERE, "Launcher ABORT due to exception:", e);
 
-            return 1;
-        }
+            launcher.launch();
+
+            return ExitCode.NORMAL;
+        } catch (IllegalArgumentException e) {
+            Log.fatal("Illegal argument", e);
 
-        // Main loop
-        if (action == Action.Abort) {
-            LOGGER.info("Launch aborted by the launcher.");
+            return ExitCode.ILLEGAL_ARGUMENT;
+        } catch (Throwable e) {
+            Log.fatal("Exception caught from launcher", e);
 
-            return 1;
+            return ExitCode.ERROR;
         }
+    }
 
-        try {
-            Launcher launcher =
-                    LauncherFactory
-                            .getInstance()
-                            .createLauncher(params);
+    private static PreLaunchAction parseLine(String input, Parameters params) throws ParseException {
+        switch (input) {
+            case "":
+                break;
 
-            launcher.launch();
+            case "launch":
+                return PreLaunchAction.LAUNCH;
 
-            return 0;
-        } catch (IllegalArgumentException e) {
-            LOGGER.log(Level.SEVERE, "Wrong argument.", e);
+            case "abort":
+                return PreLaunchAction.ABORT;
 
-            return 1;
-        } catch (Exception e) {
-            LOGGER.log(Level.SEVERE, "Exception caught from launcher.", e);
+            default:
+                String[] pair = StringUtils.splitStringPair(' ', input);
```

Comment:
```suggestion
                String[] pair = input.split(" ", 2);
```

Reply:
This is just reverting the change that was made for performance. However, if it was changed to " " instead of "\\s+" that probably would still improve performance.

Reply:
Yeah, but I don't really see why we would need to have our own function here, as we should keep the code concise. I think we have been using split for the last few years without anyone caring, so I don't think the performance gain is worth the extra code.